policies_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2015 The gocql Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gocql
+
+import "testing"
+
+func TestRoundRobinHostPolicy(t *testing.T) {
+	policy := NewRoundRobinHostPolicy()
+
+	hosts := []HostInfo{
+		HostInfo{HostId: "0"},
+		HostInfo{HostId: "1"},
+	}
+
+	policy.SetHosts(hosts)
+
+	// the first host selected is actually at [1], but this is ok for RR
+	iter := policy.Pick(nil)
+	if actual := iter(); actual != &hosts[1] {
+		t.Errorf("Expected hosts[0] but was hosts[%s]", actual.HostId)
+	}
+	if actual := iter(); actual != &hosts[0] {
+		t.Errorf("Expected hosts[1] but was hosts[%s]", actual.HostId)
+	}
+	iter = policy.Pick(nil)
+	if actual := iter(); actual != &hosts[0] {
+		t.Errorf("Expected hosts[0] but was hosts[%s]", actual.HostId)
+	}
+	if actual := iter(); actual != &hosts[1] {
+		t.Errorf("Expected hosts[1] but was hosts[%s]", actual.HostId)
+	}
+	iter = policy.Pick(nil)
+	if actual := iter(); actual != &hosts[1] {
+		t.Errorf("Expected hosts[0] but was hosts[%s]", actual.HostId)
+	}
+	if actual := iter(); actual != &hosts[0] {
+		t.Errorf("Expected hosts[1] but was hosts[%s]", actual.HostId)
+	}
+}
+
+func TestTokenAwareHostPolicy(t *testing.T) {
+	policy := NewTokenAwareHostPolicy(NewRoundRobinHostPolicy())
+
+	hosts := []HostInfo{
+		HostInfo{HostId: "0", Peer: "0", Tokens: []string{"00"}},
+		HostInfo{HostId: "1", Peer: "1", Tokens: []string{"25"}},
+		HostInfo{HostId: "2", Peer: "2", Tokens: []string{"50"}},
+		HostInfo{HostId: "3", Peer: "3", Tokens: []string{"75"}},
+	}
+
+	policy.SetHosts(hosts)
+	policy.SetPartitioner("OrderedPartitioner")
```

Comment:
Would it be a good idea to have test coverage for `Murmur3Partitioner` and `RandomPartitioner` (in addition to `OrderedPartitioner` as implemented here), or is that covered somewhere else?